qiskit/pulse/schedule.py

```diff
@@ -0,0 +1,182 @@
+        """Return duration of supplied channels.
+
+        Args:
+            *channels: Supplied channels
+        """
+        return self.timeslots.ch_duration(*channels)
+
+    def ch_start_time(self, *channels: List[Channel]) -> int:
+        """Return minimum start time for supplied channels.
+
+        Args:
+            *channels: Supplied channels
+        """
+        return self.timeslots.ch_start_time(*channels)
+
+    def ch_stop_time(self, *channels: List[Channel]) -> int:
+        """Return maximum start time for supplied channels.
+
+        Args:
+            *channels: Supplied channels
+        """
+        return self.timeslots.ch_stop_time(*channels)
+
+    def union(self, *schedules: List[ScheduleComponent]) -> 'Schedule':
+        """Return a new schedule which is the union of `self` and `schedule`.
+
+        Args:
+            *schedules: Schedules to be take the union with the parent `Schedule`.
+        """
+        return ops.union(self, *schedules)
+
+    def shift(self: ScheduleComponent, time: int) -> 'Schedule':
+        """Return a new schedule shifted forward by `time`.
+
+        Args:
+            time: Time to shift by
+        """
+        return ops.shift(self, time)
+
+    def insert(self, start_time: int, schedule: ScheduleComponent) -> 'Schedule':
+        """Return a new schedule with `schedule` inserted within `self` at `start_time`.
+
+        Args:
+            start_time: time to be inserted
+            schedule: schedule to be inserted
+        """
+        return ops.insert(self, start_time, schedule)
+
+    def append(self, schedule: ScheduleComponent) -> 'Schedule':
+        """Return a new schedule with `schedule` inserted at the maximum time over
+        all channels shared between `self` and `schedule`.
+
+        Args:
+            schedule: schedule to be appended
+        """
+        return ops.append(self, schedule)
+
+    def flatten(self, time: int = 0) -> Iterable[Tuple[int, ScheduleComponent]]:
+        """Iterable for flattening Schedule tree.
+
+        Args:
+            time: Shifted time due to parent
+
+        Yields:
+            Tuple[int, ScheduleComponent]: Tuple containing time `ScheduleComponent` starts
+                at and the flattened `ScheduleComponent`.
+        """
+        for child in self.children:
+            yield from child.flatten(time + self._shift_time)
+
+    def __add__(self, schedule: ScheduleComponent) -> 'Schedule':
+        """Return a new schedule with `schedule` inserted within `self` at `start_time`."""
+        return self.append(schedule)
+
+    def __or__(self, schedule: ScheduleComponent) -> 'Schedule':
+        """Return a new schedule which is the union of `self` and `schedule`."""
+        return self.union(schedule)
+
+    def __lshift__(self, time: int) -> 'Schedule':
+        """Return a new schedule which is shifted forward by `time`."""
+        return self.shift(time)
+
+    def __rshift__(self, time: int) -> 'Schedule':
```

Comment:
I think supporting only `__lshift__` (not supporting `__rshift__` asymmetrically) makes sense, because not a few novice users will just see `<<` as "at" and won't remember it is `<<` or `>>`. They will accidentally use `>>` intending `<<`. I'd like to give priority to avoiding user's misusage rather than completing operator definitions. Honestly speaking, I myself sometimes use `>>` intending `<<`. I know `<<` is a shift "up", but I often wrongly recognize `<<` as moving thing toward left, i.e. shifting time toward negative.

Reply:
Ok I will remove. It is always easier to add this in the future than to take away from users.